spec/02-integration/05-proxy/09-websockets_spec.lua

```diff
@@ -38,6 +38,57 @@ for _, strategy in helpers.each_strategy() do
       return wc
     end
 
+    local function check_headers(host, port)
+      local encode_base64 = ngx.encode_base64
+      local rand = math.random
+      local char = string.char
+      local sock = assert(ngx.socket.tcp())
+      local host = helpers.get_proxy_ip(false)
+      local port = helpers.get_proxy_port(false)
+
+      assert(sock:connect(host, port))
+
+      local bytes = char(rand(256) - 1, rand(256) - 1, rand(256) - 1,
+        rand(256) - 1, rand(256) - 1, rand(256) - 1,
+        rand(256) - 1, rand(256) - 1, rand(256) - 1,
+        rand(256) - 1, rand(256) - 1, rand(256) - 1,
+        rand(256) - 1, rand(256) - 1, rand(256) - 1,
+        rand(256) - 1)
+
+      local key = encode_base64(bytes)
+      local req = "GET /up-ws HTTP/1.1\r\nUpgrade: websocket\r\nHost: "
+        .. host .. ":" .. port
+        .. "\r\nSec-WebSocket-Key: " .. key
+        .. "\r\nSec-WebSocket-Version: 13"
+        .. "\r\nConnection: Upgrade\r\n\r\n"
+
+      assert(sock:send(req))
+
+      local header_reader = sock:receiveuntil("\r\n\r\n")
```

Comment:
Can we close the socket once we are done here?

Reply:
Sure. Thanks.